include/deal.II/sundials/sunlinsol_newempty.h

```diff
@@ -0,0 +1,146 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2020 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//-----------------------------------------------------------
+
+/*
+ * This file contains a verbatim copy of code snippets distributed within the
+ * SUNDIALS package, see the license below.
+ */
+
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel Reynolds @ SMU
+ *                David J. Gardner, Carol S. Woodward, and
+ *                Slaven Peles @ LLNL
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2020, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+
+  BSD 3-Clause License
```

Comment:
I believe we do not typically include the full legal note of TPLs in the main `include` folder of deal.II, especially not in the present case where the code is so trivial that we cannot be accused for copyright violation. I think it would be enough to refer to the SUNDIALS license here (web address) plus possible the authors listed above. Given that I suggest some changes below, I think that is more than fair.